python-package/xgboost/sklearn.py

```diff
@@ -536,12 +537,16 @@ def fit(self, X, y, sample_weight=None, base_margin=None,
             else:
                 params.update({'eval_metric': eval_metric})
 
-        self._Booster = train(params, train_dmatrix,
-                              self.get_num_boosting_rounds(), evals=evals,
-                              early_stopping_rounds=early_stopping_rounds,
-                              evals_result=evals_result, obj=obj, feval=feval,
-                              verbose_eval=verbose, xgb_model=xgb_model,
-                              callbacks=callbacks)
+        try:
```

Comment:
What's the reason for the try/catch?

Reply:
Scikit Learn doesn't recognize XBoost error.  Exception catching is a part of its interface.